### ISSUE TITLE
The line between the source list and folder view sometimes disappears

### ISSUE THREAD

**Issue by ryonakano** [elementary/files#617]
![peek 2018-09-14 14-49](https://user-images.githubusercontent.com/26003928/45532081-a0771d00-b82d-11e8-8d3d-f7e4d5c255ac.gif)

It may be due to my system environment, but the line between the source list and folder view sometimes disappears.
I'm on an Core i7-2630QM with NVIDIA GeForce GT 540M. I installed NVIDIA binary driver version 340.106 for GPU from AppCenter.

**Comment by tkashkin** [comment]
It also appears in other apps with `Gtk.Paned` widget, so it's a theme bug.

**Comment by cassidyjames** [comment]
@tkashkin or possibly a GTK bug; I don't see why the stylesheet would differ in NVIDIA versus Intel graphics other than due to a bug in GTK.

**Comment by cassidyjames** [comment]
(I do also see this when using NVIDIA graphics)

**Comment by tkashkin** [comment]
Interesting, I don't see this with Intel graphics too. But I also don't see anything similar with other GTK themes on NVIDIA.

**Comment by jeremypw** [comment]
I do not think this can be fixed in the Files code so I am marking as Out of Scope

**Comment by bernardodsanderson** [comment]
I was able to replicate this by installing the nvidia-driver-390 from the appcenter. Before that I had no issues. Running NVIDIA GeForce GTX 950, Juno 5.0.

**Comment by danila-io** [comment]
> I was able to replicate this by installing the nvidia-driver-390 from the appcenter. Before that I had no issues. Running NVIDIA GeForce GTX 950, Juno 5.0.

What drivers did you use prior to installing 390 from the AppCenter?

**Comment by bernardodsanderson** [comment]
It was just a clean install, so whatever it comes with. Unfortunately I didn't check prior to see exactly. I also tried updating to nvidia-415 latest driver and I'm still seeing this issue.

**Comment by bernardodsanderson** [comment]
Another update. I tested the app "Ping!" which uses `Gtk.paned`. It gives the same issues as "Files" on Elementary OS, however, when I tried it on PopOS with Nvidia drivers I did not get the same issue.

**Comment by danila-io** [comment]
@bernardodsanderson I heard that System76 distribute Nvidia drivers differently. So it's either that or it's something specific to elementary OS due to its theme.

**Comment by bernardodsanderson** [comment]
Thanks, @danila-io. I did some more digging. The issue seems to come from using a `linear-gradient` for a `background-image` to `paned.horizontal > separator` (in the `gtk-widgets.css` file). When I used a solid color for a background I did not get this issue. I also noticed when debugging that when using the "Rendering Mode" to "Image" also fixed this issue.

A separate point to make is that I noticed that this issue only appears when there's no scroll-sidebar in the main pane. When there's a lot of files and you have to scroll to see them all, there's no problem.

```
paned.horizontal > separator {
    background-image:
        linear-gradient(
            to right,
            @menu_separator 1px,
            @menu_separator_shadow 1px,
            @menu_separator_shadow 2px,
            transparent 2px
        );
    margin-right: -7px;
    min-width: 8px;
}
```

![files_issue](https://user-images.githubusercontent.com/5623786/50496842-9ce05e00-09f8-11e9-8cc7-8bd9cd570ca1.png)

**Comment by danila-io** [comment]
@bernardodsanderson What tool did you use to change rendering mode? Sorry, I'm a bit new to this stuff.

**Comment by bernardodsanderson** [comment]
@danila-io no problem, I'm pretty new too! I used  `GTK_DEBUG=interactive io.elementary.files` to get the debug dialog.

**Comment by Manueljlin** [comment]
For some reason, this issue doesn't happen in hidpi mode

**Comment by Manueljlin** [comment]
I fixed it with this on gtk-widgets.css:
```
paned.horizontal > separator {
    background-image:
        linear-gradient(
            to right,
            @menu_separator 1px,
            @menu_separator_shadow 1px,
            @menu_separator_shadow 2px,
            transparent 2px
        );
    margin-right: 0px;
    min-width: 1px;
}
```

The problem is that it's a bit harder to resize and it's only on horizontal dividers.
